pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -534,6 +534,19 @@ static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
       String comparisonCol = tableConfig.getUpsertConfig().getComparisonColumn();
       Preconditions.checkState(schema.hasColumn(comparisonCol), "The comparison column does not exist on schema");
     }
+    }
+    Preconditions.checkState(isAggregateMetricsEnabled(tableConfig),
+                "Metrics aggregation and upsert cannot be enabled together");
+  }
+
+  /**
+      * Checks if Metrics aggregation is enabled.
+      */
+  @VisibleForTesting
+     static boolean isAggregateMetricsEnabled(TableConfig config) {
+      if(config.getIndexingConfig() == null || config.getIngestionConfig() == null) {return false;
+      }
+      return config.getIndexingConfig().isAggregateMetrics() && CollectionUtils.isEmpty(config.getIngestionConfig().getAggregationConfigs());
```

Comment:
This should be `||` instead of `&&`

Reply:
I think I referred `MutableSegmentImpl` and misunderstood. Fixed it.